### PR TITLE
Alert Rule activations model

### PR DESCRIPTION
**Pull request by nhsiehgit**
per [step5](https://www.notion.so/sentry/Alert-Threshold-Implementation-Details-Activated-Alert-Rules-f1e9eaa6c26d41b4898b95f3c8e4411a?pvs=4#261c6d34d5704bc2b517377cb2d436c1)

set up a table to start recording the activated alert rule results to

### CODE REVIEW

File: src/sentry/incidents/models/alert_rule.py [hunk 1]
```diff
@@ -435,6 +435,31 @@ class Meta:
         unique_together = (("alert_rule", "label"),)
 
 
+@region_silo_only_model
+class ActivatedAlertRuleResults(Model):
+    """
+    This model represents the monitor results for an activated Alert Rule.
```
[the remark this answers is not in the document]
Comment (nhsiehgit) [reply src/sentry/incidents/models/alert_rule.py]: Oof - different

Updated my changes 😅

File: src/sentry/incidents/models/alert_rule_activations.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    objects = AlertRuleActivationsManager()
+
+    alert_rule = FlexibleForeignKey("sentry.AlertRule", related_name="activations")
+    # date_added timestamp indicates when this particular run was activated
+    date_added = models.DateTimeField(default=timezone.now)
+    # If finished_ts is null, this indicates whether the run is ongoing or completed
+    finished_ts = models.DateTimeField(null=True)
```
Comment (nhsiehgit) [inline src/sentry/incidents/models/alert_rule_activations.py:36]: maybe `finished_at` instead?

although `_ts` may be nice to denote a timestamp? 🤔

**Review by saponifi3d — APPROVED** [review]
👍 makes sense to me, you might want to have someone else take a quick look too just to make sure i didn't miss any newbie things.

File: src/sentry/incidents/models/alert_rule_activations.py (hunk 2 above)
Comment (saponifi3d) [reply src/sentry/incidents/models/alert_rule_activations.py:36]: yeah - i would make sure `finished_at` and `date_added` are consistent with other models (`created_at` a thing here?)

File: src/sentry/incidents/logic.py [hunk 3]
```diff
@@ -946,6 +947,21 @@ def create_alert_rule_activation_condition(
     return condition
 
 
+def create_alert_rule_activation(
+    alert_rule: AlertRule,
+    metric_value: int | None = 100,
```
Comment (wedamija) [inline src/sentry/incidents/logic.py:952]: Feels a little weird that we have a default here. I assume it's for tests, you should probably do the defaulting there

**Review by wedamija — APPROVED** [review]

**Pull request merged**